tests/SixLabors.Core.Tests/Helpers/DebugGuardTests.cs

```diff
@@ -0,0 +1,260 @@
+            });
+
+            Assert.Equal("myParamName", exception.ParamName);
+            Assert.True(exception.Message.Contains($"Value must be greater than {min}."));
+        }
+
+        [Theory]
+        [InlineData(2, 1)]
+        [InlineData(1, 1)]
+        public void MustBeGreaterThanOrEqualTo_IsGreaterOrEqual_ThrowsNoException(int value, int min)
+        {
+            DebugGuard.MustBeGreaterThanOrEqualTo(value, min, "myParamName");
+        }
+
+        [Fact]
+        public void MustBeGreaterThanOrEqualTo_IsLess_ThrowsNoException()
+        {
+            var exception = Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                DebugGuard.MustBeGreaterThanOrEqualTo(1, 2, "myParamName");
+            });
+
+            Assert.Equal("myParamName", exception.ParamName);
+            Assert.True(exception.Message.Contains($"Value must be greater than or equal to 2."));
+        }
+
+        [Fact]
+        public void IsTrue_IsTrue_ThrowsNoException()
+        {
+            DebugGuard.IsTrue(true, "myParamName", "myTestMessage");
+        }
+
+        [Fact]
+        public void IsTrue_IsFalse_ThrowsException()
+        {
+            var exception = Assert.Throws<ArgumentException>(() =>
+            {
+                DebugGuard.IsTrue(false, "myParamName", "myTestMessage");
+            });
+
+            Assert.Equal("myParamName", exception.ParamName);
+            Assert.True(exception.Message.Contains("myTestMessage"));
+        }
+
+        [Fact]
+        public void IsFalse_IsFalse_ThrowsNoException()
+        {
+            DebugGuard.IsFalse(false, "myParamName", "myTestMessage");
+        }
+
+        [Fact]
+        public void IsFalse_IsTrue_ThrowsException()
+        {
+            var exception = Assert.Throws<ArgumentException>(() =>
+            {
+                DebugGuard.IsFalse(true, "myParamName", "myTestMessage");
+            });
+
+            Assert.Equal("myParamName", exception.ParamName);
+            Assert.True(exception.Message.Contains("myTestMessage"));
+        }
+
+        [Theory]
+        [InlineData(new int[] { 1, 2 }, 1)]
+        [InlineData(new int[] { 1, 2 }, 2)]
+        public void MustBeSizedAtLeast_LengthIsGreaterOrEqual_ThrowsNoException(int[] value, int minLength)
+        {
+            DebugGuard.MustBeSizedAtLeast<int>(value, minLength, "myParamName");
+        }
+
+        [Fact]
+        public void MustBeSizedAtLeast_LengthIsLess_ThrowsException()
+        {
+            var exception = Assert.Throws<ArgumentException>(() =>
+            {
+                DebugGuard.MustBeSizedAtLeast<int>(new int[] { 1, 2 }, 3, "myParamName");
+            });
+
+            Assert.Equal("myParamName", exception.ParamName);
+            Assert.Contains($"The size must be at least 3.", exception.Message);
+        }
+
+        [Fact]
+        public void ReadOnlySpan_MustBeSizedAtLeast_LengthIsLess_ThrowsException()
```

Comment:
I would call this method: `MustBeSizedAtLeastReadOnlySpan_LengthIsLess_ThrowsException` or `MustBeSizedAtLeast_ReadOnlySpan_LengthIsLess_ThrowsException`. At least start with the method name. I can rename them after the merge if you would like that.